### PR TITLE
[PLAT-6905]  Interpolator with 0 values. Option test with 0 volatility.

### PR DESCRIPTION
**Pull request by marc-opengamma**
Improve SquareLinear and TimeSquare interpolator to support 0 values. Added an example of STIR futures with 0 volatility in E2E test.
TimeSquareInterpolator1D: coverage is 100%
SquareLinearInterpolator1D: coverage is 96.4%

### CODE REVIEW

File: projects/OG-Analytics/src/main/java/com/opengamma/analytics/math/interpolation/SquareLinearInterpolator1D.java [hunk 1]
```diff
@@ -76,14 +89,32 @@ public double firstDerivative(final Interpolator1DDataBundle data, final Double
...
+    }
+    if ((y2 < EPS) && (y1 >= EPS) && (x2 - value) < EPS) { // On one vertex with value 0, other vertex not 0
+      throw new OpenGammaRuntimeException("ask for first derivative on value without derivative");
+    }
+    if ((y2 < EPS) && (y1 < EPS) && ((x2 - value) < EPS || (value - x1) < EPS)) {
+      throw new OpenGammaRuntimeException("ask for first derivative on value without derivative");
+    }
     double w2 = (x2 - value) / (x2 - x1);
-    double w1 = 1.0 - w2;
+    if ((y2 < EPS) && (y2 < EPS)) {
```
Comment (yukiiwashita) [inline projects/OG-Analytics/src/main/java/com/opengamma/analytics/math/interpolation/SquareLinearInterpolator1D.java]: the two conditions are identical

File: projects/OG-Analytics/src/main/java/com/opengamma/analytics/math/interpolation/SquareLinearInterpolator1D.java [hunk 2]
```diff
@@ -76,14 +89,32 @@ public double firstDerivative(final Interpolator1DDataBundle data, final Double
     }
     double x2 = boundedValues.getHigherBoundKey();
     double y2 = boundedValues.getHigherBoundValue();
+    if ((y1 < EPS) && (y2 >= EPS) && (value - x1) < EPS) { // On one vertex with value 0, other vertex not 0
+      throw new OpenGammaRuntimeException("ask for first derivative on value without derivative");
```
Comment (yukiiwashita) [inline projects/OG-Analytics/src/main/java/com/opengamma/analytics/math/interpolation/SquareLinearInterpolator1D.java]: Change message specific to node sensitivities.

File: projects/OG-Analytics/src/main/java/com/opengamma/analytics/math/interpolation/SquareLinearInterpolator1D.java [hunk 3]
```diff
@@ -76,14 +89,32 @@ public double firstDerivative(final Interpolator1DDataBundle data, final Double
...
+    }
+    if ((y2 < EPS) && (y1 >= EPS) && (x2 - value) < EPS) { // On one vertex with value 0, other vertex not 0
+      throw new OpenGammaRuntimeException("ask for first derivative on value without derivative");
```
Comment (yukiiwashita) [inline projects/OG-Analytics/src/main/java/com/opengamma/analytics/math/interpolation/SquareLinearInterpolator1D.java]: Change message specific to node sensitivities.

File: projects/OG-Analytics/src/main/java/com/opengamma/analytics/math/interpolation/SquareLinearInterpolator1D.java [hunk 4]
```diff
@@ -76,14 +89,32 @@ public double firstDerivative(final Interpolator1DDataBundle data, final Double
...
+    }
+    if ((y2 < EPS) && (y1 < EPS) && ((x2 - value) < EPS || (value - x1) < EPS)) {
+      throw new OpenGammaRuntimeException("ask for first derivative on value without derivative");
```
Comment (yukiiwashita) [inline projects/OG-Analytics/src/main/java/com/opengamma/analytics/math/interpolation/SquareLinearInterpolator1D.java]: Change message specific to node sensitivities.

**Comment by marc-opengamma** [comment]
Corrected SquareLinear interpolator close to 0.
Change exception message to be more explicit.
Coverage is now:
TimeSquareInterpolator1D: 100%
SquareLinearInterpolator1D: 100%

File: projects/OG-Analytics/src/main/java/com/opengamma/analytics/math/interpolation/TimeSquareInterpolator1D.java [hunk 5]
```diff
@@ -65,8 +71,8 @@ public double firstDerivative(final Interpolator1DDataBundle data, final Double
 
   @Override
   public double[] getNodeSensitivitiesForValue(final Interpolator1DDataBundle data, final Double value) {
-    Validate.notNull(value, "Value to be interpolated must not be null");
-    Validate.notNull(data, "Data bundle must not be null");
+    ArgumentChecker.notNull(value, "Value to be interpolated must not be null");
+    ArgumentChecker.notNull(data, "Data bundle must not be null");
```
Comment (yukiiwashita) [inline projects/OG-Analytics/src/main/java/com/opengamma/analytics/math/interpolation/TimeSquareInterpolator1D.java]: Add ArgumentChecker for value, as done in the other methods.

**Pull request merged**